slahmr/run_rerun_vis.py

```diff
@@ -0,0 +1,331 @@
+    num_faces = world_smpl["faces"].shape[-2]
+
+    # faces has no batch dim here, use expand to avoid copy
+    meshes = pytorch3d.structures.Meshes(
+        verts=world_smpl["vertices"].reshape(num_meshes, num_vertices, 3),
+        faces=world_smpl["faces"].expand(num_meshes, num_faces, 3),
+    )
+
+    vertices = world_smpl["vertices"].numpy(force=True)
+    faces = world_smpl["faces"].numpy(force=True)
+    vertex_normals = (
+        meshes.verts_normals_padded()
+        .reshape(num_tracks, num_frames, num_vertices, 3)
+        .numpy(force=True)
+    )
+
+    # NOTE: if the meshes don't deform over time or are similar we could use the same
+    #  vertex normals for all frames and/or tracks
+
+    for frame_id in range(num_frames):
+        rr.set_time_sequence(f"frame_id_{phase_label}", frame_id)
+        rr.set_time_sequence("frame_id", frame_id)
+        translation = phase_result["cam_t"][1, frame_id].numpy(force=True)
+        rotation_mat = phase_result["cam_R"][1, frame_id].numpy(force=True)
+        rotation_q = transform.Rotation.from_matrix(rotation_mat).as_quat()
+        rr.log_rigid3(
+            f"world/{phase_label}/camera",
+            child_from_parent=(translation, rotation_q),
+            xyz="RDF",
+        )
+        for i, _ in enumerate(dataset.track_ids):
+            if vis_mask[i][frame_id] >= 0:
+                rr.log_mesh(
+                    f"world/{phase_label}/#{i}",
+                    vertices[i, frame_id],
+                    indices=faces,
+                    normals=vertex_normals[i, frame_id],
+                    albedo_factor=_index_to_color(i),
+                )
+            else:
+                rr.log_cleared(
+                    f"world/{phase_label}/#{i}",
+                )
+    rr.set_time_sequence(f"frame_id_{phase_label}", None)
+
+
+def log_input_frames(dataset: dataset.MultiPeopleDataset, phase_label: str) -> None:
+    """Log raw input video to rerun."""
+    for frame_id, img_path in enumerate(dataset.sel_img_paths):
+        rr.set_time_sequence(f"frame_id_{phase_label}", frame_id)
+        rr.set_time_sequence("frame_id", frame_id)
+        rr.log_image_file(f"world/{phase_label}/camera/image", img_path=img_path)
+    rr.set_time_sequence(f"frame_id_{phase_label}", None)
+
+
+def log_skeleton_2d(dataset: dataset.MultiPeopleDataset, phase_label: str) -> None:
+    """Log 2D skeleton to rerun."""
+    skeleton_ids = np.array(
+        [
+            [15, 13],
+            [13, 11],
+            [16, 14],
+            [14, 12],
+            [11, 12],
+            [5, 11],
+            [6, 12],
+            [5, 6],
+            [5, 7],
+            [6, 8],
+            [7, 9],
+            [8, 10],
+            [1, 2],
+            [0, 1],
+            [0, 2],
+            [1, 3],
+            [2, 4],
+            [3, 5],
+            [4, 6],
+        ]
+    )
+    idcs = [0, 16, 15, 18, 17, 5, 2, 6, 3, 7, 4, 12, 9, 13, 10, 14, 11]
```

Comment:
Break out as constants. Also, where do they come from?

Reply:
Added some comments, and capitalized.